### PR TITLE
log: don't log error for context.Canceled

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: coap-gateway/service/clientLogAndWriteErrorResponse.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 
 func (client *Client) logAndWriteErrorResponse(err error, code codes.Code, token message.Token) {
 	if err != nil {
-		log.Errorf("%v", err)
+		log.LogAndReturnError(err)
```
Comment (Danielius1922) [inline coap-gateway/service/clientLogAndWriteErrorResponse.go]: returning the error seems unnecessary

File: pkg/log/log.go [hunk 2]
```diff
@@ -114,18 +120,8 @@ func Warnf(template string, args ...interface{}) {
...
-// logger then panics. (See DPanicLevel for details.)
-func DPanicf(template string, args ...interface{}) {
-	Get().DPanicf(template, args...)
-}
-
-// Panicf uses fmt.Sprintf to log a templated message, then panics.
-func Panicf(template string, args ...interface{}) {
-	Get().Panicf(template, args...)
+	err := fmt.Errorf(template, args...)
+	LogAndReturnError(err)
```
Comment (Danielius1922) [inline pkg/log/log.go]: errcheck complains here

File: coap-gateway/service/decodeMsgToDebug.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ func decodeMsgToDebug(client *Client, resp *pool.Message, tag string) {
 			case message.AppCBOR, message.AppOcfCbor:
 				s, err := cbor.ToJSON(body)
 				if err != nil {
-					log.Errorf("Cannot encode %v to JSON: %v", body, err)
+					log.Errorf("Cannot encode %v to JSON: %w", body, err)
```
Comment (ondrejtomcik) [inline coap-gateway/service/decodeMsgToDebug.go]: keep the same style, lower case.

File: pkg/log/log.go (hunk 2 above)
Comment (ondrejtomcik) [reply pkg/log/log.go]: on many places here

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**